main/distributed-programming.md

```diff
@@ -190,113 +191,96 @@ objects in the current vat or presences for objects in other vats as arguments.
+notifier is provided to clients, while the updater is retained internally. The udpater has three
+methods, `updateState(state)`, `finish(finalState)`, and `fail(reason)`. `updateState` supplies a
+new state that will be used to resolve all the outstanding promises. `finish` and `fail` both close
+the stream, one with a final state, the other by rejecting the promise, so that clients' Promises
+will also be rejected.
+
+`notifier.getUpdateSince()` returns `{ value, updateCount }`.
+- `value` represents the state, and the format is up to the publisher.
+- `updateCount` is used to request notification the next time there's a change to the state. If the
+  state becomes final (e.g. a seat exits), updateCount will be `undefined`.  If there's an error,
+  the promise for the record is rejected and there isn't a next state.
+
+If you call `getUpdateSince(oldUpdateCount)`:
+- With no count, or any `updateCount` other than the most recent one:
+  - The notifier immediately returns a promise for a record with the current state.
+- With the most-recently generated `updateCount`:
+  - The notifier returns a promise for the next record, which is resolved on the next state change.
+- If you haven't called `getUpdateSince()` before, you won't have a previous updateCount to use.
+
+Some notification systems also provide access to a complete history of an object's state
+changes. The Agoric Notifier API only directly supports the single state change notification
+style. The client can't work around this by keeping lists of changes, since the service doesn't
+guarantee that clients will see all the changes.
+
+An alternative approach for services that want to ensure their clients receive a more complete
+history is to represent the state as a set of changes leading up to the present. Use cases for this
+include an editor with an undo function, or an application with rollback ability.
+
+Rather than sending `"the current state is 'blue'."`, a contract could send `"the current state is
+'blue', the most recent update was { ''blah' => 'blue' }"`.  If you, as a contract author,
+determine that clients want this much detail, you would have to package and send it.
+
+### Follower pattern
+
+A common pattern for following updates to a notifier until it's done is the following.  Note that
+the notifier object is remote, and so we use `E()`.  Also, `PublicAPI` is a widely available
+contract facet, where it often makes sense to put a `getNotifier()` method.
+
+<<< @/snippets/test-distributed-programming.js#follower
 
-Zoe supports the Notifier, which publishes updates to offer state
-(reallocations and completions). Some contracts also use it, and
-can publish current prices or other contract-specific details.
+### Providing updates
 
-### Getting notifications
+Objects that want to share info about changes to their state can use a notifier to provide
+updates. Publishers import and call `produceNotifier()`, which returns two facets, a notifier
```

Comment:
Where is `produceNotifier` coming from? The snippet below doesn't match.

Reply:
This is a holdover.

I didn't attempt to put all the inline snippets into the, which means this kind of thing can happen. Is there a way to cite a small snippet in-line, or is there only the `<<<` approach, which  boxes the citation in a formatted block.

Reply:
I think there's no way to do an in-line snippet import